src/sentry/static/sentry/app/utils/discover/fields.tsx

```diff
@@ -261,83 +261,385 @@ export type Aggregation = {
+  },
+  [FieldKey.DEVICE_LOCALE]: {
+    key: 'device.locale',
+    name: 'device.locale',
+  },
+  [FieldKey.DEVICE_UUID]: {
+    key: 'device.uuid',
+    name: 'device.uuid',
+  },
+  [FieldKey.DEVICE_ARCH]: {
+    key: 'device.arch',
+    name: 'device.arch',
+  },
+  [FieldKey.DEVICE_BATTERY_LEVEL]: {
+    key: 'device.battery_level',
+    name: 'device.battery_level',
+  },
+  [FieldKey.DEVICE_ORIENTATION]: {
+    key: 'device.orientation',
+    name: 'device.orientation',
+  },
+  [FieldKey.DEVICE_SIMULATOR]: {
+    key: 'device.simulator',
+    name: 'device.simulator',
+  },
+  [FieldKey.DEVICE_ONLINE]: {
+    key: 'device.online',
+    name: 'device.online',
+  },
+  [FieldKey.DEVICE_CHARGING]: {
+    key: 'device.charging',
+    name: 'device.charging',
+  },
+  [FieldKey.GEO_COUNTRY_CODE]: {
+    key: 'geo.country_code',
+    name: 'geo.country_code',
+  },
+  [FieldKey.GEO_REGION]: {
+    key: 'geo.region',
+    name: 'geo.region',
+  },
+  [FieldKey.GEO_CITY]: {
+    key: 'geo.city',
+    name: 'geo.city',
+  },
+  [FieldKey.ERROR_TYPE]: {
+    key: 'error.type',
+    name: 'error.type',
+  },
+  [FieldKey.ERROR_VALUE]: {
+    key: 'error.value',
+    name: 'error.value',
+  },
+  [FieldKey.ERROR_MECHANISM]: {
+    key: 'error.mechanism',
+    name: 'error.mechanism',
+  },
+  [FieldKey.ERROR_HANDLED]: {
+    key: 'error.handled',
+    name: 'error.handled',
+  },
+  [FieldKey.STACK_ABS_PATH]: {
+    key: 'stack.abs_path',
+    name: 'stack.abs_path',
+  },
+  [FieldKey.STACK_FILENAME]: {
+    key: 'stack.filename',
+    name: 'stack.filename',
+  },
+  [FieldKey.STACK_PACKAGE]: {
+    key: 'stack.package',
+    name: 'stack.package',
+  },
+  [FieldKey.STACK_MODULE]: {
+    key: 'stack.module',
+    name: 'stack.module',
+  },
+  [FieldKey.STACK_FUNCTION]: {
+    key: 'stack.function',
+    name: 'stack.function',
+  },
+  [FieldKey.STACK_IN_APP]: {
+    key: 'stack.in_app',
+    name: 'stack.in_app',
+  },
+  [FieldKey.STACK_COLNO]: {
+    key: 'stack.colno',
+    name: 'stack.colno',
+  },
+  [FieldKey.STACK_LINENO]: {
+    key: 'stack.lineno',
+    name: 'stack.lineno',
+  },
+  [FieldKey.STACK_STACK_LEVEL]: {
+    key: 'stack.stack_level',
+    name: 'stack.stack_level',
+  },
+  [FieldKey.TRANSACTION_DURATION]: {
+    key: 'transaction.duration',
+    name: 'transaction.duration',
+  },
+  [FieldKey.TRANSACTION_OP]: {
+    key: 'transaction.op',
+    name: 'transaction.op',
+  },
+  [FieldKey.TRANSACTION_STATUS]: {
+    key: 'transaction.status',
+    name: 'transaction.status',
+  },
+  [FieldKey.TRACE]: {
+    key: 'trace',
+    name: 'trace',
+  },
+  [FieldKey.TRACE_SPAN]: {
+    key: 'trace.span',
+    name: 'trace.span',
+  },
+  [FieldKey.TRACE_PARENT_SPAN]: {
+    key: 'trace.parent_span',
+    name: 'trace.parent_span',
+  },
+  [FieldKey.PROJECT]: {
+    key: 'project',
+    name: 'project',
+  },
+  [FieldKey.ISSUE]: {
+    key: 'issue',
+    name: 'issue',
+  },
+}) as Readonly<Record<FieldKey, FieldTag>>;
```

Comment:
Is there not a way to generate this from the `FIELDS` and `FieldKey` definitions?

Reply:
👍 to generating this